torch/_refs/__init__.py

```diff
@@ -1539,6 +1539,24 @@ def sub(
 )
 
 
+@register_decomposition(torch.ops.aten.xlogy)
+@out_wrapper()
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a", "b"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+)
+def xlogy(a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]):
+    assert isinstance(a, TensorLike) or isinstance(b, TensorLike)
+    if isinstance(a, TensorLike) and isinstance(b, Number):
+        b = scalar_tensor(b, dtype=a.dtype, device=a.device)
+    elif isinstance(b, TensorLike) and isinstance(a, Number):
+        a = scalar_tensor(a, dtype=b.dtype, device=b.device)
+    assert isinstance(a, TensorLike)
```

Comment:
I think these asserts can be removed

Reply:
lint complains without the asserts.

Reply:
It's OK then to add a comment explaining the asserts are to satisfy the linter, or to disable the linter's checks on the line that's complaining (with a note about why the check is disabled)